### PR TITLE
Docs: useSignal(), userStore()

### PR DESCRIPTION
**Pull request by un33k**
# What is it?

- [ x ] Docs

Streamline the useSignal and useStore section of state

# Use cases and why

Clean up the docs a bit, starting with useSignal and useStore.

# Checklist:

- [ x ] My code follows the [developer guidelines of this project](https://github.com/BuilderIO/qwik/blob/main/CONTRIBUTING.md)
- [ x ] I have performed a self-review of my own code

### CODE REVIEW

**Comment by manucorporat** [comment]
I am not convinced by some of the changes in this PR. We are trying to write docs that help AI to give better answers, some of the new docs add some noise to what useSignal is about. Will try next monday to explain in detail what i mean.

**Comment by manucorporat** [comment]
In fact i would love to simplify this docs even more, to talk about `Reactive State` only, maybe some little note at the bottom about static state and serialization

**Review by manucorporat — CHANGES_REQUESTED** [review]
Let's discusse it before merging

**Comment by un33k** [comment]
All of my changes was created by me and finalized by GPT-4.

File: packages/docs/src/routes/docs/(qwik)/components/state/index.mdx [hunk 1]
```diff
@@ -22,24 +22,23 @@ contributors:
...
-It is important to notice that state in Qwik is not necessarily a local component state, but rather an app state that can be instantiated by any component.
+  - Reactive state is created using the `useSignal()` or `useStore()` functions, (aka hooks).
+  - Reactive states can be shared with other components across the application to update them in response to changes.
 
 ## `useSignal()`
 
-Use `useSignal()` to create a reactive signal (a form of state). The `useSignal()` takes an initial value and returns a reactive signal.
-
-The reactive signal returned by `useSignal()` consists of an object with a single property `.value`. If you change the `value` property of the signal, any component that depends on it will be updated automatically.
+The `useSignal()` hook accepts an `initial value`, creates, and returns a `reactive state`, also known as a `signal`. This signal is an object with a single property, `.value`. Multiple components can consume the signal. When any component with access to the signal modifies the `.value` property, all dependent components will automatically update in response to the change.
```
Comment (manucorporat) [inline packages/docs/src/routes/docs/(qwik)/components/state/index.mdx:41]: wondering if we can split this paragram into two, seems like it's talking about several concepts! we are finding that keeping paragraphs focus on a single concept can boost GPT results

Comment (un33k) [reply packages/docs/src/routes/docs/(qwik)/components/state/index.mdx]: Yeah, I agree, the split look much better and to-the-point.

**Comment by un33k** [comment]
@manucorporat in pursuit of PR reduction, and with the knowledge and the trust that you folks will make the documentation better and AI digestible,  I am closing this PR to reduce the noise.

**Pull request closed without merging**